src/dialog.jsx

```diff
@@ -128,14 +127,19 @@ let Dialog = React.createClass({
       autoDetectWindowHeight: false,
       autoScrollBodyContent: false,
       actions: [],
-      modal: false,
       repositionOnUpdate: true,
     };
   },
 
   getInitialState() {
+    let open = (this.props.defaultIsOpen || this.props.openImmediately || false);
```

Comment:
I would move this in an else condition after `// isOpen overwrites any default that may have been set`